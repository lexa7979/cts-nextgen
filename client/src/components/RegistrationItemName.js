/* eslint-disable react/prop-types */
/* eslint-disable no-use-before-define */

import React from "react";
import * as yup from "yup";

import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";

/* eslint-disable no-template-curly-in-string */
const propsSchema = yup.object( {
	formikBag: yup.object()
		.shape( {
			values:	      yup.object()
				.required()
				.test( "is-object", "${path}: object expected", value => value instanceof Object ),
			handleChange: yup.object()
				.required()
				.test( "is-function", "${path}: callback expected", value => typeof value === "function" ),
			errors:       yup.object()
				.required()
				.test( "is-object", "${path}: object expected", value => value instanceof Object ),
		} )
		.required(),
} );
/* eslint-enable no-template-curly-in-string */

export default RegistrationItemName;

/**
 * Renders a component which contains element to let
 * the user input his/her name.
 *
 * @param	{object}	props
 *
 * @returns	{object}	React component
 */
function RegistrationItemName( props ) {
	propsSchema.strict().validateSync( props );	// eslint-disable-line no-sync

	const { values, handleChange, errors } = props.formikBag;

	return (
		<Grid item container direction="row" spacing={1}>
			<Grid item xs={12} sm={6}>
				<TextField
					id="firstname"
					value={values.firstname}
					label="Firstname"
					onChange={handleChange}
					margin="normal"
					variant="outlined"
					fullWidth
					required
					error={Boolean( errors.firstname )}
					helperText={errors.firstname}
				/>
			</Grid>
			<Grid item xs={12} sm={6}>
				<TextField
					id="lastname"
					value={values.lastname}
					label="Lastname"
					onChange={handleChange}
					margin="normal"
					variant="outlined"
					fullWidth
					required
					error={Boolean( errors.lastname )}
					helperText={errors.lastname}
				/>
			</Grid>
		</Grid>
	);
}
