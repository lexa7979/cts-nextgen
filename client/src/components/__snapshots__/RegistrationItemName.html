
<!DOCTYPE html>
<html style="height: 100%;">
  <head>
  </head>
  <body style="display: flex; flex-flow: column nowrap; justify-content: center; align-items: center; height: 100%;">
    <h2>
      Component RegistrationItemName - when rendering into snapshots with valid properties - delivers expected result  (-> check snapshot)
    </h2>
    <!--[ Content from unit test: ]-->
    <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item">
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6">
        <div class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth">
          <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                 data-shrink="false"
                 for="firstname"
                 id="firstname-label"
          >
            Firstname
            <span class="MuiFormLabel-asterisk MuiInputLabel-asterisk">
               
              *
            </span>
          </label>
          <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl">
            <input type="text"
                   aria-invalid="false"
                   id="firstname"
                   required
                   value
                   class="MuiInputBase-input MuiOutlinedInput-input"
            >
            <fieldset aria-hidden="true"
                      style="padding-left:8px"
                      class="PrivateNotchedOutline-root-327 MuiOutlinedInput-notchedOutline"
            >
              <legend class="PrivateNotchedOutline-legend-328"
                      style="width:0.01px"
              >
                <span>
                  &#8203;
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6">
        <div class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth">
          <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                 data-shrink="false"
                 for="lastname"
                 id="lastname-label"
          >
            Lastname
            <span class="MuiFormLabel-asterisk MuiInputLabel-asterisk">
               
              *
            </span>
          </label>
          <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl">
            <input type="text"
                   aria-invalid="false"
                   id="lastname"
                   required
                   value
                   class="MuiInputBase-input MuiOutlinedInput-input"
            >
            <fieldset aria-hidden="true"
                      style="padding-left:8px"
                      class="PrivateNotchedOutline-root-327 MuiOutlinedInput-notchedOutline"
            >
              <legend class="PrivateNotchedOutline-legend-328"
                      style="width:0.01px"
              >
                <span>
                  &#8203;
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <!--[ End of included content ]-->
    <br>
    <br>
    &lt;RegistrationItemName formikBag={exampleFormikBag} /&gt;
  </body>
</html>
